File: opcua-webservice-spring/examples/uawebservicetest/src/main/java/org/opcfoundation/uawebservicetest/testdigitaltwin/EnumTestDataType.java
package org.opcfoundation.uawebservicetest.testdigitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.webserver.addressspace.models.UaEnumDataType;
import org.opcfoundation.webserver.digitaltwin.DigitalTwinSpace;

import java.util.List;

public class EnumTestDataType extends UaEnumDataType {
    EnumTestDataType(DigitalTwinSpace space)
    {
        super(
                "EnumTest",
                new LocalizedText("EnumTest"),
                List.of("A","B","C","D"),
                space);
    }
}

File: opcua-webservice-node/source/test/testdigitaltwin/EnumTestDataType.ts
import { UaLocalizedText } from "opcua-webapi-ts";
import { UaEnumDataType,DigitalTwinSpace } from "../../src";

export class EnumTestDataType extends UaEnumDataType {
    constructor(space: DigitalTwinSpace) {
        super(
            "EnumTest",
            new UaLocalizedText("EnumTest"),
            ["A", "B", "C", "D"],
            space,
        );

        this.description = new UaLocalizedText("EnumTestDataType");
    }
}